workflow/automation/lib/MgmtDB.py

```diff
@@ -4,18 +4,29 @@
 import os
 import sqlite3 as sql
 from typing import List, Union
+from dataclasses import dataclass
 
 import qcore.constants as const
 from qcore.qclogging import get_basic_logger
 
 
 Process = const.ProcessType
 
-SchedulerTask = namedtuple(
-    "SchedulerTask", ["run_name", "proc_type", "status", "job_id", "error"]
-)
-# Make error an optional value, once we are using Python 3.7 then this can be made nicer..
-SchedulerTask.__new__.__defaults__ = (None,)
+
+@dataclass
+class SchedulerTask:
```

Comment:
Do we want to give this class to/from json methods, so the only place json is read/written is in here?